tests/functional/pyocf/rio.py

```diff
@@ -0,0 +1,287 @@
+                    iodir,
+                    0,
+                    0,
+                )
+                io.set_data(data)
+                io.callback = self.get_io_cb()
+                self.ios += self.jobspec.bs
+                io.submit()
+                with self.qd_condition:
+                    self.qd += 1
+
+            with self.qd_condition:
+                self.qd_condition.wait_for(lambda: self.qd == 0)
+
+    def __init__(self):
+        self.global_jobspec = JobSpec()
+        self.jobs = []
+
+        self._threads = []
+        self.errors = {}
+        self.error_count = 0
+
+    def readwrite(self, rw: ReadWrite):
+        self.global_jobspec.readwrite = rw
+        return self
+
+    def rwmixwrite(self, mix: int):
+        self.global_jobspec.rwmixwrite = mix
+        return self
+
+    def rwmixread(self, mix: int):
+        self.global_jobspec.rwmixwrite = 100 - mix
+        return self
+
+    def norandommap(self):
+        self.global_jobspec.randommap = False
+        return self
+
+    def bs(self, bs: Size):
+        self.global_jobspec.bs = bs
+        return self
+
+    def offset(self, offset: Size):
+        self.global_jobspec.offset = offset
+        return self
+
+    def njobs(self, njobs: int):
+        self.global_jobspec.njobs = njobs
+        return self
+
+    def qd(self, qd: int):
+        self.global_jobspec.qd = qd
+        return self
+
+    def target(self, target: Volume):
+        self.global_jobspec.target = target
+        return self
+
+    def add_job(self, job: JobSpec):
+        self.jobs.append(job)
+        return self
+
+    def size(self, size: Size):
+        self.global_jobspec.size = size
+        return self
+
+    def io_size(self, size: Size):
+        self.global_jobspec.io_size = size
+        return self
+
+    def time_based(self):
+        self.global_jobspec.time_based = True
+        return self
+
+    def time(self, time: timedelta):
+        self.global_jobspec.time = time
+        return self
+
+    def continue_on_error(self):
+        self.global_jobspec.continue_on_error = True
+        return self
+
+    def abort(self):
+        for thread in self._threads:
+            thread.abort()
+
+        self.wait_for_completion()
+        return self
+
+    def wait_for_completion(self):
+        for thread in self._threads:
+            thread.join()
+            self.errors.update({thread.name: thread.errors})
+            self.error_count += len(thread.errors)
+
+        return self
+
+    def __del__(self):
+        self.wait_for_completion()
+
+    def clear(self):
+        self._threads = []
+        self.errors = {}
+
+    def run(self):
+        self.run_async()
+        self.wait_for_completion()
+        return self
+
+    def run_async(self):
+        self.clear()
+
+        jobs = deepcopy(self.jobs)
+
+        if not jobs:
+            for _ in range(self.global_jobspec.njobs):
+                jobs.append(self.global_jobspec)
```

Comment:
The loop seems unnecessary. How about `jobs = [self.global_jobspec] * self.global_jobspec.njobs`?